src/main/java/org/fenixedu/academic/domain/Degree.java

```diff
@@ -390,6 +390,25 @@ public boolean isDegreeOrBolonhaDegreeOrBolonhaIntegratedMasterDegree() {
         return getDegreeType().isDegreeOrBolonhaDegreeOrBolonhaIntegratedMasterDegree();
     }
 
+    public static Degree find(final String code) {
+        if (StringUtils.isBlank(code)) {
+            return null;
+        }
+
+        Degree degreeFound = null;
+        for (Degree degree : Degree.readNotEmptyDegrees()) {
+            if (StringUtils.equalsIgnoreCase(degree.getCode(), code)) {
+                if (degreeFound != null) {
```

Comment:
Why are you checking for duplicates here? Doing this check on EVERY read operation seems like a huge waste of computing time, for an issue that this method has nothing to do with.